cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/TimePartitionedFileSet.java

```diff
@@ -62,25 +63,39 @@
 
   /**
    * Adds a new metadata entry for a particular partition.
-   * Note that existing entries can not be updated.
+   * If the metadata key already exists, it will be overwritten.
    *
    * @param time the partition time in milliseconds since the Epoch
-   *
-   * @throws DataSetException in case an attempt is made to update existing entries.
+   * @throws PartitionNotFoundException when a partition for the given time is not found
```

Comment:
I spoke with @chtyim in person.
He mentioned that we should have another method `setMetadata` that allows mutability, while maintaining the behavior for `setMetadata`. We can decide whether or not to deprecate the `setMetadata` later.